plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -299,6 +455,39 @@ private Optional<DataFileInfo> rewriteParquetFile(Path path, ImmutableLongBitmap
         return Optional.of(fileWriter.getDataFileInfo());
     }
 
+    private void storeCdfEntries(Page page, int[] deleted, int deletedCount, FileDeletion deletion)
+    {
+        if (cdfEnabled) {
+            if (cdfPageSink == null) {
+                cdfPageSink = cdfPageSinkSupplier.get();
+            }
+            Page cdfPage = page.getPositions(deleted, 0, deletedCount);
```

Comment:
can `cdfPage.getPositionCount()` be 0 at this point?

Reply:
it can be 0

Reply:
i think it would be good to exit early when it is. 
even before `cdfPageSink = cdfPageSinkSupplier.get();`